### PR TITLE
Expose node agent INGRESS_GATEWAY_FALLBACK_SECRET and SECRET_WATCHER_RESYNC_PERIOD as Helm configurations

### PR DESCRIPTION
**Pull request by Monkeyanator**
Expose `SECRET_WATCHER_RESYNC_PERIOD` and `INGRESS_GATEWAY_FALLBACK_SECRET` as Helm configurations.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Review by incfly — COMMENTED** [review]
@howardjohn is this the correct place to make changes, w.r.t new installer? or somewhere else?

**Comment by howardjohn** [comment]
What do we need these for? Users can set these without being in values.yaml and they are already documented as env vars on istio.io

**Comment by Monkeyanator** [comment]
These configurations aren't documented [here](https://istio.io/docs/reference/config/installation-options/#nodeagent-options) like `nodeagent.env.CA_PROVIDER` and `nodeagent.env.CA_ADDR` are. Might be that we could just change the docs, but the per-component docs are generated from the `values.yaml` comments.

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
@istio/wg-security-maintainers please take a look

File: install/kubernetes/helm/istio/charts/nodeagent/values.yaml [hunk 1]
```diff
@@ -4,12 +4,18 @@
...
+  # Name of authentication provider.
   CA_PROVIDER: ""
   # CA endpoint.
   CA_ADDR: ""  
-  # names of authentication provider's plugins.
+  # Names of authentication provider's plugins.
   PLUGINS: ""
+  # The agent will retrieve secrets from the API server with the frequency
+  # specified in this configuration.
+  SECRET_WATCHER_RESYNC_PERIOD: ""
```
Comment (incfly) [inline install/kubernetes/helm/istio/charts/nodeagent/values.yaml]: I think the "" is not a good idea. I know we have code in binary to handle if time.Duration parse error, we use as some hardcode value.

I would rather make this to match against the default values in code and then make hardcoded go in node agent simpler.

File: install/kubernetes/helm/istio/charts/nodeagent/values.yaml [hunk 2]
```diff
@@ -4,12 +4,18 @@
...
+  # When a secret cannot be found with the expected suffix in the ingress namespace
+  # a secret with this name will be used instead as a fallback.
+  INGRESS_GATEWAY_FALLBACK_SECRET: "gateway-fallback"
```
Comment (incfly) [inline install/kubernetes/helm/istio/charts/nodeagent/values.yaml]: quick code search, I found this this ends up in `secretfetcher.go:132`, and then we'll use this name for fallback.

@JimmyCYJ could you confirm? Want to make sure this is the secret name, not the in-line secret data.

**Pull request closed without merging**